src/api/c/device.cpp

```diff
@@ -74,11 +74,10 @@ af_err af_get_active_backend(af_backend *result)
 af_err af_init()
 {
     try {
-        static bool first = true;
-        if(first) {
-            getDeviceInfo();
-            first = false;
-        }
+        static std::once_flag flag;
+        std::call_once(flag, []() {
```

Comment:
I would think `std::call_once(flag, getDeviceInfo)` would work here. no need for a lambda.